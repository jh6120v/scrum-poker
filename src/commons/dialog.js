import React from 'react';
import PropTypes from 'prop-types';
import {
    CancelButton, ConfirmButton, DialogContent, DialogFooter, DialogShadow, DialogWrap
} from '../styles/dialog-style';

const Dialog = ({ message, onCancel, onConfirm }) => (message ? (
    <>
        <DialogShadow />
        <DialogWrap>
            <DialogContent>{message}</DialogContent>
            <DialogFooter>
                <CancelButton onClick={onCancel}>Cancel</CancelButton>
                <ConfirmButton onClick={onConfirm}>Confirm</ConfirmButton>
            </DialogFooter>
        </DialogWrap>
    </>
) : null);

Dialog.propsTypes = {
    message: PropTypes.string.isRequired,
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired
};

export default Dialog;