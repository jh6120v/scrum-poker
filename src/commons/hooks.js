import React, { useState, useCallback } from 'react';
import {
    ConfirmButton, ModelContent, ModelFooter, ModelShadow, ModelWrap
} from '../styles/model-style';

const useModel = (message, confirm) => {
    const [isShown, setShown] = useState(false);
    const showModal = useCallback(() => setShown(true), []);
    const hideModal = useCallback(() => setShown(false), []);

    const ModelBox = () => (
        <div>
            <ModelWrap>
                <ModelContent>{message}</ModelContent>
                <ModelFooter>
                    <ConfirmButton onClick={confirm}>Confirm</ConfirmButton>
                </ModelFooter>
            </ModelWrap>
            <ModelShadow onClick={hideModal} />
        </div>
    );

    return {
        ModelBox,
        isShown,
        showModal,
        hideModal
    };
};

export {
    useModel
};
