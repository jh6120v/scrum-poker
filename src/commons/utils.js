import React, { Suspense } from 'react';

export const createActionCreator = (namespace) => (actionType) => {
    const type = `${namespace}/${actionType}`;
    const actionCreator = (payload = {}) => ({
        type,
        payload
    });

    actionCreator.type = type;
    Object.freeze(actionCreator);

    return actionCreator;
};

export const waitingRouteComponent = (Component) => () => (
    <Suspense fallback={<div>載入中...</div>}>
        <Component />
    </Suspense>
);
