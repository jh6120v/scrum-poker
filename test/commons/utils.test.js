import { lazy } from 'react';
import { render, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createActionCreator, waitingRouteComponent } from '../../src/commons/utils';

describe('test utils', () => {
    it('test createActionCreator type', () => {
        const actionCreator = createActionCreator('@@NAMESPACE');
        const test = actionCreator('TEST');

        expect(test.type).toEqual('@@NAMESPACE/TEST');
    });

    it('test createActionCreator func', () => {
        const actionCreator = createActionCreator('@@NAMESPACE');
        const test = actionCreator('TEST');

        const fakeTestCreator = () => ({
            type: '@@NAMESPACE/TEST',
            payload: {}
        });

        expect(JSON.stringify(test())).toEqual(JSON.stringify(fakeTestCreator()));
    });

    it('rendered lazily fallback', async () => {
        const LazyComponent = lazy(() => import('../lazy-component'));

        const { container } = render(
            waitingRouteComponent(LazyComponent)()
        );

        const lazyElement = await waitForElement(() => container);

        expect(lazyElement).toMatchSnapshot();
    });

    it('rendered lazily', async () => {
        const LazyComponent = lazy(() => import('../lazy-component'));

        const { getByText } = render(
            waitingRouteComponent(LazyComponent)()
        );

        const lazyElement = await waitForElement(() => getByText(/i am lazy/i));
        expect(lazyElement).toBeInTheDocument();
    });
});
