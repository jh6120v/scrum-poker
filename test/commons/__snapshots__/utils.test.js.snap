// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test utils rendered lazily fallback 1`] = `
<div>
  <div>
    載入中...
  </div>
</div>
`;
